Sources/ClientRuntime/Networking/Http/SdkHttpRequest.swift

```diff
@@ -70,18 +70,19 @@ extension SdkHttpRequest: CustomDebugStringConvertible, CustomStringConvertible
 }
 
 extension SdkHttpRequestBuilder {
-    public func update(from crtRequest: HTTPRequest, originalRequest: SdkHttpRequest) -> SdkHttpRequestBuilder {
+    public func update(from crtRequest: HTTPRequestBase, originalRequest: SdkHttpRequest) -> SdkHttpRequestBuilder {
         headers = convertSignedHeadersToHeaders(crtRequest: crtRequest)
         methodType = originalRequest.method
         host = originalRequest.endpoint.host
-        let pathAndQueryItems = URLComponents(string: crtRequest.path)
-        path = pathAndQueryItems?.path ?? "/"
-        queryItems = pathAndQueryItems?.percentEncodedQueryItems ?? [URLQueryItem]()
-
+        if let crtRequest = crtRequest as? HTTPRequest {
+            let pathAndQueryItems = URLComponents(string: crtRequest.path)
+            path = pathAndQueryItems?.path ?? "/"
+            queryItems = pathAndQueryItems?.percentEncodedQueryItems ?? [URLQueryItem]()
+        }
```

Comment:
Should we do something instead of silently no-oping when the `crtRequest` is not a HTTPRequest? 

Would it make sense to to still set the `path` and `queryItems` since we have handling for nil values? Or perhaps we should fail? 

```suggestion
        let pathAndQueryItems = (crtRequest as? HTTPRequest).map { URLComponents(string: $0.path) }
        path = pathAndQueryItems?.path ?? "/"
        queryItems = pathAndQueryItems?.percentEncodedQueryItems ?? [URLQueryItem]()
```

Reply:
Was thinking around this for sometime but the current way seems correct.

We have 

https://github.com/awslabs/aws-crt-swift/blob/0.5.6/Source/AwsCommonRuntimeKit/http/HTTPRequestBase.swift#L7

The  H1 and H2 has their own implementations

H1

https://github.com/awslabs/aws-crt-swift/blob/0.5.6/Source/AwsCommonRuntimeKit/http/HTTPRequest.swift#L8

H2 

https://github.com/awslabs/aws-crt-swift/blob/0.5.6/Source/AwsCommonRuntimeKit/http/HTTPRequest.swift#L62

What we must do always if the passed object is neither HTTPRequest nor HTTP2Request, it fails. This is a developer error here.